src/components/Aframe/my-loader.js

```diff
@@ -432,141 +432,85 @@ AFRAME.registerComponent("myloader", {
-      colorTransfer = new Uint8Array(3 * 256);
-      for (var i = 0; i < 256; i++) {
-        colorTransfer[i * 3] = colorData[i * 4];
-        colorTransfer[i * 3 + 1] = colorData[i * 4 + 1];
-        colorTransfer[i * 3 + 2] = colorData[i * 4 + 2];
-      }
-      iam.colorTransfer = colorTransfer;
-      iam.updateTransfertexture();
-    };
-    this.colorMap.img.src = imgColorImage.src;
+    if (!this.colorTransferMap.has(this.currentColorMap)) {
+      var colorCanvas = document.createElement("canvas");
+      var iam = this;
+      let newColorMap = {
+        img: document.createElement("img"),
+        width: 255,
+        height: 15,
+        data: null,
+      };
+      var imgWidth = newColorMap.width;
+      var imgHeight = newColorMap.height;
+
+      newColorMap.img.src = this.currentColorMap;
+      this.colorTransferMap.set(this.currentColorMap, newColorMap);
+      const mappedColorMap = newColorMap;
+      newColorMap.img.onload = function (data) {
+        colorCanvas.height = imgHeight;
+        colorCanvas.width = imgWidth;
+        var colorContext = colorCanvas.getContext("2d");
+        colorContext.drawImage(newColorMap.img, 0, 0);
+        var colorData = colorContext.getImageData(0, 0, imgWidth, 1).data;
+        var colorTransfer = new Uint8Array(3 * 256);
+        for (var i = 0; i < 256; i++) {
+          colorTransfer[i * 3] = colorData[i * 4];
+          colorTransfer[i * 3 + 1] = colorData[i * 4 + 1];
+          colorTransfer[i * 3 + 2] = colorData[i * 4 + 2];
+        }
+        mappedColorMap.data = colorTransfer;
+        iam.updateTransferTexture();
+      };
+    } else {
+      this.updateTransferTexture();
+    }
   },
 
   update: function (oldData) {
-    if (
-      oldData.cameraState !== undefined &&
-      oldData.cameraState !== this.data.cameraState
-    ) {
+    if (oldData === undefined) {
+      return;
     }
 
+    // this part updates the opacity control points
     if (
-      oldData.channel !== undefined &&
-      oldData.channel !== this.data.channel
+      (this.data.alphaXDataArray !== undefined &&
+        oldData.alphaXDataArray !== this.data.alphaXDataArray) ||
+      (this.data.alphaYDataArray !== undefined &&
+        oldData.alphaYDataArray !== this.data.alphaYDataArray)
     ) {
-      if (this.el.getObject3D("mesh") !== undefined) {
-        let material = this.el.getObject3D("mesh").material;
-        material.uniforms.u_lut.value = null;
-        material.uniforms.useLut.value = false;
-        material.uniforms.channel.value = this.data.channel;
-        material.needsUpdate = true;
-      }
-      return;
+      this.updateOpacityData(
+        this.data.alphaXDataArray,
+        this.data.alphaYDataArray
+      );
+      this.updateTransferTexture();
     }
 
     if (
-      oldData.colorMapping !== undefined &&
-      oldData.colorMapping !== this.data.colorMapping
+      oldData.path !== this.data.path ||
+      oldData.colorMap !== this.data.colorMap
```

Comment:
I think the `loadModel` handles the updating the color map - is that correct? If not (or maybe either way for efficiency reasons), should this be a separate if statement for the color map change check, which just calls `this.updateTransferTexture()`?

Reply:
I applied your suggestion. Split the check for new colormap from the new model.
This may require re test after Robert merges